pkg/cri/instrument/instrumented_service.go

```diff
@@ -90,7 +93,9 @@ func (in *instrumentedService) checkInitialized() error {
 // GRPC service request handlers should return error before server is fully
 // initialized.
 // NOTE(random-liu): All following functions MUST check initialized at the beginning.
-func (in *instrumentedAlphaService) checkInitialized() error {
+func (in *instrumentedAlphaService) checkInitialized(ctx context.Context) error {
+	log.G(ctx).Warning("CRI API v1alpha2 is deprecated since containerd v1.7 and removed in containerd v2.0. Use CRI API v1 instead.")
+	in.warn.Emit(ctx, deprecation.CRIAPIV1Alpha2)
```

Comment:
It looks like this will log _every_ time a v1alpha2 RPC is invoked?  We probably want to limit this somehow, otherwise the containerd logs will be filled with warnings (and we'll have a lot of locking inside `warn.Emit` that could impact performance).

A `sync.Once` could be useful to do that.

Reply:
makes sense. Done